libs/core/src/lib/avatar-group/components/avatar-group-host.component.ts

```diff
@@ -0,0 +1,154 @@
+import { NgIf } from '@angular/common';
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ContentChildren,
+    DestroyRef,
+    ElementRef,
+    inject,
+    Input,
+    OnChanges,
+    QueryList,
+    signal,
+    SimpleChanges,
+    ViewEncapsulation
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { HasElementRef, ResizeObserverDirective } from '@fundamental-ngx/cdk/utils';
+import { animationFrames, combineLatest, delayWhen, map, Observable, startWith, Subject } from 'rxjs';
+import { AvatarGroupItemRendererDirective } from '../directives/avatar-group-item-renderer.directive';
+import { AvatarGroupItemDirective } from '../directives/avatar-group-item.directive';
+import { AvatarGroupHostConfig } from '../types';
+
+@Component({
+    selector: 'fd-avatar-group-host',
+    host: {
+        class: 'fd-avatar-group',
+        '[class.fd-avatar-group--individual-type]': 'type === "individual"',
+        '[class.fd-avatar-group--group-type]': 'type === "group"',
+        '[class.fd-avatar-group--horizontal]': 'orientation === "horizontal"',
+        '[class.fd-avatar-group--vertical]': 'orientation === "vertical"',
+        '[class.fd-avatar-group--xs]': 'size === "xs"',
+        '[class.fd-avatar-group--s]': 'size === "s"',
+        '[class.fd-avatar-group--m]': 'size === "m"',
+        '[class.fd-avatar-group--l]': 'size === "l"',
+        '[class.fd-avatar-group--xl]': 'size === "xl"'
```

Comment:
In `AvatarGroupOverflowButtonComponent` you use NgClass directive to compose complex classList, here you use host. Any particular reason to use different style?